great_expectations/experimental/datasources/spark_s3_datasource.py

```diff
@@ -0,0 +1,155 @@
+from great_expectations.experimental.datasources.spark_datasource import (
+    SparkDatasourceError,
+)
+from great_expectations.experimental.datasources.spark_file_path_datasource import (
+    CSVAsset,
+)
+
+if TYPE_CHECKING:
+    from botocore.client import BaseClient
+
+    from great_expectations.experimental.datasources.interfaces import (
+        BatchSorter,
+        BatchSortersDefinition,
+    )
+
+
+logger = logging.getLogger(__name__)
+
+
+BOTO3_IMPORTED = False
+try:
+    import boto3
+
+    BOTO3_IMPORTED = True
+except ImportError:
+    pass
+
+
+class SparkS3DatasourceError(SparkDatasourceError):
+    pass
+
+
+class SparkS3Datasource(_SparkFilePathDatasource):
+    # instance attributes
+    type: Literal["spark_s3"] = "spark_s3"
+
+    # S3 specific attributes
+    bucket: str
+    boto3_options: Dict[str, Any] = {}
+
+    _s3_client: Union[BaseClient, None] = pydantic.PrivateAttr(default=None)
+
+    def _get_s3_client(self) -> BaseClient:
+        s3_client: Union[BaseClient, None] = self._s3_client
+        if not s3_client:
+            # Validate that "boto3" libarary was successfully imported and attempt to create "s3_client" handle.
+            if BOTO3_IMPORTED:
+                try:
+                    s3_client = boto3.client("s3", **self.boto3_options)
+                except Exception as e:
+                    # Failure to create "s3_client" is most likely due invalid "boto3_options" dictionary.
+                    raise SparkS3DatasourceError(
+                        f"""Due to exception: "{str(e)}", "s3_client" could not be created."""
+                    ) from e
+            else:
+                raise SparkS3DatasourceError(
+                    """Unable to create "SparkS3Datasource" due to missing boto3 dependency."""
+                )
+
+            self._s3_client = s3_client
+
+        return s3_client
+
+    def test_connection(self, test_assets: bool = True) -> None:
+        """Test the connection for the SparkS3Datasource.
+
+        Args:
+            test_assets: If assets have been passed to the SparkS3Datasource, whether to test them as well.
+
+        Raises:
+            TestConnectionError: If the connection test fails.
+        """
+        try:
+            _ = self._get_s3_client()
+        except Exception as e:
+            raise TestConnectionError(
+                "Attempt to connect to datasource failed with the following error message: "
+                f"{str(e)}"
+            ) from e
+
+        if self.assets and test_assets:
+            for asset in self.assets.values():
+                asset.test_connection()
+
+    def add_csv_asset(
+        self,
+        name: str,
+        batching_regex: Optional[Union[str, re.Pattern]] = None,
+        header: bool = False,
+        infer_schema: bool = False,
+        prefix: str = "",
+        delimiter: str = "/",
+        max_keys: int = 1000,
+        order_by: Optional[BatchSortersDefinition] = None,
+    ) -> CSVAsset:
```

Comment:
`non-blocking`
I think it's okay for a followup because this PR is about getting the S3 Spark datasource working but... aren't there more arguments that can be passed to spark's equivalent of `read_csv()`?

Reply:
@Kilo59 We should discuss it; so far, we have not those use cases, but if we can come up with them, then the supporting arguments should definitely be added.  (Or we can go back and re-add "kwargs".)